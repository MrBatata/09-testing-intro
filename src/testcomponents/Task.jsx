import React, { useState } from 'react';

const Task = (props) => {

  const [checked, setchecked] = useState(props.task.isCompleted)

  const handleOnChange = () => {
    setchecked(!checked)
  }

  return (
    <div data-testid={`task-test-${props.task.id}`}>
      <h3>Hola</h3>
      <div>
        <span>ID: {props.task.id}</span>
        <span>Descripci√≥n: {props.task.description}</span>
      </div>
      <div>
        <label name="completed">Completado</label>
        <input name="completed" id="checkbox"
          type='checkbox'
          onChange={handleOnChange}
          checked={checked}
        />
      </div>
    </div>
  );
}

export default Task;
