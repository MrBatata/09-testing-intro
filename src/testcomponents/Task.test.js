import React from "react";
import { render, screen, cleanup } from "@testing-library/react"
import Task from "./Task";
import renderer from "react-test-renderer";


afterEach(() => cleanup());

describe('Task component is working correctly', () => {
  const task1 = { id: 1, description: 'Tarea 1', isCompleted: true }
  render(<Task task={task1}></Task>);
  const TaskElement1 = screen.getByTestId(`task-test-${task1.id}`);
  const checkbox1 = TaskElement1.querySelector('#checkbox')

  it('should render Task component', () => {
    expect(TaskElement1).toBeInTheDocument();

  });

  it('should have right texts', () => {
    expect(TaskElement1).toHaveTextContent(task1.id);
    expect(TaskElement1).toHaveTextContent(task1.description);
  });

  it('should be checked', () => {
    expect(checkbox1).toBeChecked();
  });

  const task2 = { id: 42, description: 'Tarea 42', isCompleted: false }
  render(<Task task={task2}></Task>);
  const TaskElement2 = screen.getByTestId(`task-test-${task2.id}`);
  const checkbox2 = TaskElement2.querySelector('#checkbox');

  it('should be checked', () => {
    expect(checkbox2).not.toBeChecked();
  });

  it('matches snapshot ', () => {
    const snapTask = { id: 15, description: 'Snapshot 2', isCompleted: false }
    const tree = renderer.create(<Task task={snapTask}></Task>)
    expect(tree).toMatchSnapshot();
  });

})

