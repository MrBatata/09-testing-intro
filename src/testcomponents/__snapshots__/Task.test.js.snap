// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component is working correctly matches snapshot  1`] = `
<div
  data-testid="task-test-15"
>
  <h3>
    Hola
  </h3>
  <div>
    <span>
      ID: 
      15
    </span>
    <span>
      Descripci√≥n: 
      Snapshot 2
    </span>
  </div>
  <div>
    <label
      name="completed"
    >
      Completado
    </label>
    <input
      checked={false}
      id="checkbox"
      name="completed"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
